larreco/ClusterFinder/ClusterAna_module.cc

```diff
@@ -57,54 +76,58 @@ namespace cluster {
     TH1F* fCRE;
     TH1F* fCRP;
     // Neutrino interactions
-    TH1F* fNuKE_elec;
-    TH1F* fNuKE_muon;
-    TH1F* fNuKE_pion;
-    TH1F* fNuKE_kaon;
-    TH1F* fNuKE_prot;
-    //  TH1F* fNuEP2;
-    TH1F* fNuEP2_elec;
-    TH1F* fNuEP2_muon;
-    TH1F* fNuEP2_pion;
-    TH1F* fNuEP2_kaon;
-    TH1F* fNuEP2_prot;
-    TH1F* fNuE_elec;
-    TH1F* fNuE_muon;
-    TH1F* fNuE_pion;
-    TH1F* fNuE_kaon;
-    TH1F* fNuE_prot;
-    TH1F* fNuP_elec;
-    TH1F* fNuP_muon;
-    TH1F* fNuP_pion;
-    TH1F* fNuP_kaon;
-    TH1F* fNuP_prot;
+    TH1F* fT_elec;
+    TH1F* fT_muon;
+    TH1F* fT_pion;
+    TH1F* fT_kaon;
+    TH1F* fT_prot;
+    //  TH1F* fEP;
+    TH1F* fEP_elec;
+    TH1F* fEP_muon;
+    TH1F* fEP_pion;
+    TH1F* fEP_kaon;
+    TH1F* fEP_prot;
+    TH1F* fE_elec;
+    TH1F* fE_muon;
+    TH1F* fE_pion;
+    TH1F* fE_kaon;
+    TH1F* fE_prot;
+    TH1F* fP_elec;
```

Comment:
There are a number of places where information per particle type is used.  I suggest creating a structure that handles all of these things:
```.cpp
struct ParticleInfo {
  TH1F* fT;
  TH1F* fEP;
  TH1F* fE;
  TH1F* fP;
  TProfile* fEP_T;
  std::array<float, 2> fKERange;
};
enum pdg_index {invalid=-1, electron=0, muon, pion, kaon, proton}; 
std::array<ParticleInfo, 5> info_for_particle;
```
Then I would introduce a function that does a translation from PDGCode to index, similar to what you have:
```.cpp
pdg_index ClusterAna::PDGCodeIndex(int pdgCode)
{
  switch(abs(pdgCode)) {
  case 11: return pdg_index::electron;
  case 13: return pdg_index::muon;
  case 211: return pdg_index::pion;
  case 321: return pdg_index::kaon;
  case 2212: return pdg_index::proton;
  default: return pdg_index::invalid;
  }
}
```
With this rearrangement, things can be a bit clearer.  See below.